packages/provider/src/RelayClient.ts

```diff
@@ -469,12 +470,47 @@ export class RelayClient {
     provider,
     config = {}
   }: GSNUnresolvedConstructorInput): Promise<GSNConfig> {
+    let configFromServer = {}
+    // todo fix contractInteractor
+    let chainId = 0
+    await new Promise((resolve, reject) => {
+      provider.send({
+        jsonrpc: '2.0',
+        method: 'eth_chainId',
+        params: [],
+        id: Date.now()
+      }, (e: Error | null, r: any) => {
+        if (errorAsBoolean(e)) {
+          reject(e)
+        } else if (errorAsBoolean(r.error)) {
+          reject(r.error)
+        } else {
+          chainId = parseInt(r.result)
+          resolve(r.result)
+        }
+      })
+    })
+    const useGsnDocsConfig = config.useGsnDocsConfig != null ? config.useGsnDocsConfig : defaultGsnConfig.useGsnDocsConfig
+    if (useGsnDocsConfig) {
+      this.logger.debug('Reading config from docs')
+      configFromServer = await this._resolveConfigurationFromServer(chainId)
+    }
     return {
       ...defaultGsnConfig,
+      ...configFromServer,
       ...removeNullValues(config)
     }
   }
 
+  async _resolveConfigurationFromServer (chainId: number): Promise<Partial<GSNConfig>> {
+    const httpClient = new HttpClient(new HttpWrapper(), this.logger)
+    const jsonConfig = await httpClient.getNetworkConfiguration()
```

Comment:
I am concerned that if the GSN will not be able to initialise without centralised GSN "servers" responding to a request, very quickly all integrations will do `useGsnDocsConfig: false` and the feature will be dead. 
Is there a reason we wouldn't `try/catch` and proceed without the server configuration?

Reply:
fixed